File: managed-services/week-3/web-backend.py
#******************************************************************************************
# This lambda function will act as the backend for the loadbalancer and must be in a separate
# target group. You need to ensure that the healthcheck is enabled, else the function does # NOTE:
# turn healthy. Must have the path in LB as /lambda or /lambda/*
#******************************************************************************************

def lambda_handler(event, context):
    response = {
        "statusCode": 200,
        "statusDescription": "200 OK",
        "isBase64Encoded": False,
        "headers": {
        "Content-Type": "text/html; charset=utf-8"
        }
    }

    response['body'] = """
    <html>
        <head>
            <title>Hello Lambda!</title>
            <style>
                html, body {
                margin: 0; padding: 0;
                font-family: arial; font-weight: 10; font-size: 1em;
                text-align: center;
            }
            </style>
        </head>
        <body>
            <p>Hello from Lambda!</p>
        </body>
    </html>
    """

    return response

File: managed-services/week-3/web-backend.js
//*********************************************************************************************************************
// This Lambda script will act as the backend for a application load balancer
//
// Create the LB and TG as usual but in the TG select the lambda as the backend and pick this function
// from the dropdown. Also you need to enable the healthcheck else the lambda will never turn healthy
// If you are using the path based routing in LB then use /lambda as the path. Slash is important
//*********************************************************************************************************************

//Async is needed else the lambda complains that the response is not formatted well
exports.handler = async (event) => {
    console.log('Lambda is called from LB and the payload is');
    console.log(JSON.stringify(event));
    const response = {
        statusCode: 200,
        statusDescription: "200 OK",
        isBase64Encoded: false,
        headers: {
            "Content-Type": "text/html"
        },
        body: "<h1>Response from lambda web backend</h1>"
    };
    return response;
};

